main.go

```diff
@@ -54,12 +55,29 @@ type HealthCheckWebService struct {
 // }
 
 func (hc *HealthCheckWebService) apiGetIsPrimary(w http.ResponseWriter, r *http.Request) {
+	m := r.URL.Query().Get("max_allowable_byte_lag")
+
+	max_allowable_byte_lag := int64(0)
+
+	if len(m) > 0 {
+		i, err := strconv.ParseInt(m, 10, 64)
+		if err != nil {
+			log.Fatalln("Error converting query param to int64:", err)
+		}
+		max_allowable_byte_lag = i
+	}
+
 	nodeInfo, err := hc.healthChecker.dataSource.GetNodeInfo()
 	if err != nil {
 		// Return a 500. Something bad happened.
 		panic(err)
 	}
 
+	// if byte lag exceeds max_allowable_byte_lag then return 500
+	if nodeInfo.ByteLag > max_allowable_byte_lag {
```

Comment:
What if we turned this into a method with the above logic? Something like:

```
if hc.maxAlloweableByteLagExceeded(r, nodeInfo) {
  w.WriteHeader(http.StatusServiceUnavailable)
}
```

That might help clean up the duplicate code in both api methods. It would also allow you to return `true` if the max query param wasn't set:

```
func yolasdjfiasd() {
  maxAllowableByteLagString := r.URL.Query().Get("max_allowable_byte_lag")
  if len(maxAllowableByteLagString) {
    return true
  }

  // parse the param as int and check
}
```

Last point, in that method if you do have a non-int value in your query param, you can panic in the `hc.maxAlloweableByteLagExceeded` method because it's in a request path.

Reply:
ah great idea!

Reply:
hey @film42 per your example I passed in `hc.maxAlloweableByteLagExceeded` func via `HealthCheckWebService` struct. and it works, but is there a reason to do this over referencing a function outside the struct?